upload/admin/view/template/extension/shipping/mds.twig

```diff
@@ -102,15 +102,16 @@
                     </div>
                     <br>
                     {% for key, service in services %}
-                        {% set display, surcharge=_context['shipping_mds_service_display_name_' ~ key], _context['shipping_mds_service_surcharge_' ~ key] %}
+                        {% set display, surcharge, name_key,surcharge_key = _context['shipping_mds_service_display_name_' ~ key], _context['shipping_mds_service_surcharge_' ~ key],'shipping_mds_service_display_name_' ~ key,'shipping_mds_service_surcharge_' ~ key %}
```

Comment:
I'm not too familiar with this syntax - but it seems like `display` and `surcharge` are set to the same value, and that  `name_key` and `surcharge_key` are set to the same value.
Am I correct on this?
If so - why define the extra variables?

Reply:
Hi 
I would say this are building a key- value object < shipping_mds_service_surcharge_' ~ key, context ['shipping_mds_service_display_name_' ~ key]> where one would be and identifier and another the value.
So, let us say the key on the loop would be 1 it would look like this <name_key, display> in turn would be < shipping_mds_service_surcharge_1, Overnight before 16:00> 

The same applies for the surcharge – so the keys used here are then used to identify the values on checkout.

This would be my understanding for that logic.

Reply:
Aaah, I see. That's a very weird syntax.
Thank you for the explanation.